components/org.wso2.carbon.identity.application.authenticator.basicauth/src/test/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticatorTestCase.java

```diff
@@ -303,12 +321,79 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
             }
         }).when(mockAuthnCtxt).setRememberMe(anyBoolean());
 
+        mockLog = mock(Log.class);
+        enableDebugLogs(mockLog, Boolean.parseBoolean(debugEnabled));
+        doAnswer(new Answer<Object>() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+
+                debugMsg = (String) invocation.getArguments()[0];
+                return null;
+            }
+        }).when(mockLog).debug(anyString());
+
         basicAuthenticator.processAuthenticationResponse(mockRequest, mockResponse, mockAuthnCtxt);
 
-        assertEquals(authenticatedUser.getAuthenticatedSubjectIdentifier(), dummyUserName);
+        if (userNameValue
+                != null && userNameValue.trim().length() > 0) {
+            assertEquals(authenticatedUser.getAuthenticatedSubjectIdentifier(), dummyDomainName +
+                    CarbonConstants.DOMAIN_SEPARATOR + userNameValue + "@" + "dummyTenantDomain");
+        } else {
+            assertEquals(authenticatedUser.getAuthenticatedSubjectIdentifier(), dummyUserName);
+        }
         if (Boolean.valueOf(chkRemember)) {
             assertEquals(isrememberMe, (Boolean) true);
         }
+        if (Boolean.parseBoolean(debugEnabled) && userNameUri != null && userNameUri.trim().length() > 0) {
```

Comment:
Can use StringUtils.isNotBlank, check and correct at the other similar places also

Reply:
Fixed with https://github.com/wso2-extensions/identity-local-auth-basicauth/pull/43